src/views/Map/CustomerMap.js

```diff
@@ -0,0 +1,221 @@
+
+import React from 'react'
+import { AppSwitch } from '@coreui/react'
+import { Input, Row, FormGroup } from 'reactstrap';
+
+var map, infoWindow, marker, markers, geoCodeValue;
+var google = window.google;
+
+class CustomerMap extends React.PureComponent {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      searchBox: '',
+      addressBox: '',
+      checked: true
+    }
+    this.googleMapRef = React.createRef()
+    this.initAutocomplete = this.initAutocomplete.bind(this)
+    this.handleChange = this.handleChange.bind(this)
+    this.handleMapChange = this.handleMapChange.bind(this)
+
+  }
+
+  handleMapChange(value) {
+    this.setState({
+      addressBox: value
+    })
+  }
+
+  handleChange(e) {
+    switch (e.target.id) {
+      case 'pac-input':
+        this.setState({
+          searchBox: e.target.value
+        })
+        break;
+      case 'customer_address':
+        this.setState({
+          addressBox: e.target.value
+        })
+        break;
+      case 'address_switch':
+        this.setState({
+          checked: e.target.checked
+        })
+        break;
+      default:
+        break;
+    }
+  }
+
+  componentDidMount() {
+    let googleScript = document.createElement('script');
+    // API key invalid, add yours or contact @johnerry on git
+    googleScript.src = "https://maps.googleapis.com/maps/api/js?key=AIzatSyBKHtdFymwut91lTBTQBDgltrFaXPD6TxY2g&libraries=places"
+    googleScript.async = true
+    window.document.body.appendChild(googleScript)
+    googleScript.addEventListener('load', this.initAutocomplete)
+  }
+
+  initAutocomplete() {
+    let mapToState = this.handleMapChange
+    let myLatlng = { lat: 6.60954222424663, lng: 3.370920970651687 };
+    infoWindow = new google.maps.InfoWindow();
+    map = new google.maps.Map(this.googleMapRef.current, {
+      center: myLatlng,
+      zoom: 10,
+      mapTypeId: 'roadmap'
+    });
+
+    map.addListener('click', function (mapsMouseEvent) {
+      // Close the current InfoWindow.
+      infoWindow.close();
+      // Create a new InfoWindow.
+      markers.forEach(function (marker) {
+        marker.setMap(null);
+      });
+
+      geoCode(mapsMouseEvent.latLng);
+      console.log(mapsMouseEvent.latLng.toString());
+    });
+
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(function (position) {
+        let pos = {
+          lat: position.coords.latitude,
+          lng: position.coords.longitude
+        };
+
+        markers.forEach(function (marker) {
+          marker.setMap(null);
+        });
+        markers = [];
+        // markers.push(marker);
+        if (pos.length !== undefined) {
+          geoCode(pos);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['geoCode' was used before it was defined.](https://app.codacy.com/gh/BuildForSDG/Team-064-Frontend/pullRequest?prid=5703258)